### PR TITLE
agent: allow absolute paths for NK and mTLS certificate and better logging

### PR DESCRIPTION
**Pull request by THS-on**
`rsa_keyname` and `mtls_cert` now allow also absolute paths.

The logging level for creation of the keys and mTLS certificate is moved to info.

Related #878

### CODE REVIEW

File: keylime/keylime_agent.py [hunk 1]
```diff
@@ -399,18 +399,24 @@ class CloudAgentHTTPServer(ThreadingMixIn, HTTPServer):
 
     def __init__(self, server_address, RequestHandlerClass, agent_uuid):
         """Constructor overridden to provide ability to pass configuration arguments to the server"""
+        # Find the locations for the key and mTLS certificate.
+        # Those either relative to secdir (/var/lib/keylime/secure) or absolute paths.
```
Comment (axelsimon) [inline keylime/keylime_agent.py:403]:
```suggestion
        # Find the locations for the mTLS key and certificate.
        # They are either relative to secdir (/var/lib/keylime/secure) or absolute paths.
```

File: keylime/keylime_agent.py [hunk 2]
```diff
@@ -399,18 +399,24 @@ class CloudAgentHTTPServer(ThreadingMixIn, HTTPServer):
...
         # read or generate the key depending on configuration
         if os.path.isfile(keyname):
             # read in private key
-            logger.debug("Using existing key in %s", keyname)
+            logger.info("Using existing key in %s", keyname)
             f = open(keyname, "rb")
             rsa_key = crypto.rsa_import_privkey(f.read())
         else:
-            logger.debug("key not found, generating a new one")
+            logger.info("key for U/V transport and mTLS certificate not found, generating a new one")
```
Comment (axelsimon) [inline keylime/keylime_agent.py:419]:
```suggestion
            logger.info("Key for U/V transport and mTLS certificate not found, generating a new one")
```

File: keylime/keylime_agent.py [hunk 3]
```diff
@@ -421,11 +427,11 @@ def __init__(self, server_address, RequestHandlerClass, agent_uuid):
             self.rsaprivatekey)
 
         if os.path.isfile(certname):
-            logger.debug("Using existing mTLS cert in %s", certname)
+            logger.info("Using existing mTLS cert in %s", certname)
             with open(certname, "rb") as f:
                 mtls_cert = x509.load_pem_x509_certificate(f.read())
         else:
-            logger.debug("No mTLS certificate found generating a new one")
+            logger.info("No mTLS certificate found generating a new one")
```
Comment (axelsimon) [inline keylime/keylime_agent.py:434]:
```suggestion
            logger.info("No mTLS certificate found, generating a new one")
```

**Review by axelsimon — APPROVED** [review]
Looks good to me!

**Comment by sergio-correia** [comment]
This outage is probably still open and causing the issues we see here: https://status.testing-farm.io/issues/2022-02-07-fedora-ci-outage/

**Pull request merged**